Add env module

sebthom opened the pull request.

chrisant996 commented on flexprompt_modules.lua:
```diff
@@ -744,6 +744,16 @@ local function render_duration(args)
+-- ENV MODULE:  {env:var_name:background_color:foreground_color}
+--  - var_name name of the environment variable
+local function render_env(args)
+  local env_var_name, bgcolor, fgcolor = args:match("([^:]+):?([^:]*):?(.*)")
```
The `bgcolor` and `fgcolor` variable names are misleading.  The names may be intuitive in the Rainbow style, but that's not how they work in the other styles.  That's why everything else calls them `color` and `altcolor`.

chrisant996 commented on flexprompt_modules.lua:
```diff
@@ -744,6 +744,16 @@ local function render_duration(args)
+  if env_var_name == "" then
+      return
+  end
+  return os.getenv(env_var_name) or "", bgcolor, fgcolor
```
I can see how this is useful.  Before this could be included as a built-in module, some things would need to change.
- It would need to follow the same syntax as the other modules.
- It would need a way to have label text associated with the environment variable's value.
- I also have other thoughts and concerns that aren't well-formed enough to put into writing yet (sorry).

sebthom commented:
I changed the env module as requested, except for the label part as there are endless of ways how to implement it so I'd leave that up to you.

However I added a separate **label** module, that allows you to display arbitrary text to the prompt.

So you could do:
```lua
flexprompt.settings.right_prompt = "{exit}{label:stage=PROD:color=red}"
```
or even:
```lua
stage=os.getenv("stage")
flexprompt.settings.right_prompt = "{exit}{label:stage=".. stage .."}"
```
This allows you to construct the text as you like using any supported Lua constructs. But the text is only generated once during config load.

chrisant996 commented:
> You can also derive the label to display from somewhere else:
> 
> ```lua
> stage=os.getenv("stage")
> flexprompt.settings.right_prompt = "{exit}{label:stage=".. stage ..":color=red}"
> ```
> 
> This allows you to construct the text as you like using any supported Lua constructs. But the text is only generated once during config load.

FWIW that technique is susceptible to "injection attacks" because external data is injected into a string that will be later processed.  If the external data contains certain kinds of characters it can cause the prompt to malfunction.  Strings can of course always be constructed however an author wants to, but where the content comes from and what safeguards exist can be important.
